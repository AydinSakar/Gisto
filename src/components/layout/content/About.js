import React from 'react';

export const About = () => (
  <div>About</div>
);

About.propTypes = {};

export default About;
