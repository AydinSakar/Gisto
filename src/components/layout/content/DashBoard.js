import React from 'react';
import Icon from 'components/common/Icon';

const DashBoard = () => (
  <div><Icon size="12" type="add"/> Dashboard</div>
);

DashBoard.propTypes = {};

export default DashBoard;
