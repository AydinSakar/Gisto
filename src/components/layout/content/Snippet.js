import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { get, join } from 'lodash/fp';
import * as snippetActions from 'actions/snippets';
import { removeTags } from 'utils/tags';

export class Snippet extends React.Component {
  componentDidMount() {
    this.props.getSnippet(this.props.match.params.id || this.props.snippet.id);
  }

  componentDidUpdate(prevProps) {
    if (prevProps.match.params.id !== this.props.match.params.id) {
      this.props.getSnippet(this.props.match.params.id || this.props.snippet.id);
    }
  }

  render() {
    const { snippet, match } = this.props;

    return (
      <React.Fragment>
        <strong>{ removeTags(get('description', snippet)) }</strong>
        <div>ID: { match.params.id }</div>
        <div>Tags: { join(', ', get('tags', snippet)) }</div>
        <div>Star: { get('star', snippet) ? 'true' : 'false' }</div>
        <div>Languages: { join(', ', get('languages', snippet)) }</div>
      </React.Fragment>
    );
  }
}

const mapStateToProps = (state, ownProps) => {
  const snippetId = ownProps.match.params.id;

  return {
    snippet: get(['snippets', 'snippets', snippetId], state)
  };
};

Snippet.propTypes = {
  match: PropTypes.object,
  snippet: PropTypes.object,
  getSnippet: PropTypes.func.isRequired
};

export default connect(mapStateToProps, {
  getSnippet: snippetActions.getSnippet
})(Snippet);
