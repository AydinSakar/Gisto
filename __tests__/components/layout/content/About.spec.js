import { About } from 'components/layout/content/About';

const propSetup = (props) => ({
  ...props
});

const setup = (props) => shallow(<About { ...propSetup(props) }/>);

describe('COMPONENTS - <About>', () => {
  test('render About', () => {
    const component = setup();

    expect(component).toMatchSnapshot();
  });
});
