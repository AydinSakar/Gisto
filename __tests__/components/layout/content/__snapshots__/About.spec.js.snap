// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENTS - <About> render About 1`] = `
<div>
  About
</div>
`;
